### ISSUE TITLE
Drop support for Python 3.8

### ISSUE THREAD

**Issue by StevenMaude** [opensafely-core/opensafely-cli#280]
We test with Python 3.8, and also produce the build with Python 3.8. Python 3.8 is end-of-life as of 2024-10-07.

Existing users with local setups may be using an existing Python 3.8 installation to run this software. How do we get those users migrated? The solution there may be related to #277.

Are there other places where the Python version used for OpenSAFELY CLI interact with other components? (I *think* this should be independent, but I'll prompt the question in case not.)

**Comment by bloodearnest** [comment]
Just to say, 3.9 is now EOL, so we should probably remove support for that too.

Additionally, this is keeping us from updating to the latest versions of many packages, as they no longer support 3.8 (or 3.9 in many cases).

Again, worth noting that moving to uv (#277) would be an alternative approach to solving this particular problem.

**Comment by bloodearnest** [comment]
> > Stata, R or Python OpenSAFELY currently supports Stata v16.1, Python 3.8, Python 3.10, R 4.0 and R 4.4 for statistical analysis. For security reasons, available libraries are restricted to those provided by the framework.
> 
> [Introduction – Essential](https://docs.opensafely.org/#essential)

This is correct. This is talking about the python action image. `python:v1` will be python3.8 forever.

The version of python needed to run opensafely-cli is a separate thing.

> 
> > Python version 3.7 or higher must be installed on your machine to perform these tasks.
> 
> [Install python](https://docs.opensafely.org/install-python/)

This is out of date, and should be 3.8. 3.7 was the old minimum version for cohort extractor, I suspect its never been updated.

> 
> > `pyenv install 3.10:latest`
> 
> [Python](https://docs.opensafely.org/install-python/#python)

We should probably update this to track the most recent tested release of python for opensafely-cli, which is currently [3.13](https://github.com/opensafely-core/opensafely-cli/blob/main/.github/workflows/tests.yml#L10)

**Comment by StevenMaude** [comment]
Though it makes things more complicated, we may also want to consider coinciding this change with #277. If users are forced to upgrade their installations, it might be good to also switch them to install with `uv`.

**Comment by rebkwok** [comment]
Just adding to this - the pipeline library is also still on 3.8 due to its inclusion in opensafely-cli. It's now failing to apply a security update that isn't available on 3.8.

**Comment by tomodwyer** [comment]
Blocked until we get a few more responses in the poll
